File: code-coaches/hard/digits-of-pi/digits_of_pi.c
#include <stdio.h>
#include <string.h>

char get_pi_decimal_at(int index)
{
    char pi_decimals[] = "14159265358979323846264338327950288419716939937510582097494459230781640628620899862803482534211706798214808651328230664709384460955058223172535940812848111745028410270938521105559644622948954930381964428";
    char decimal = pi_decimals[index];
    return decimal;
}

int main()
{
    // - 1 to convert the index to 0-based index
    int index;
    scanf("%d", &index);
    index--;

    printf("%c\n", get_pi_decimal_at(index));

    return 0;
}

File: code-coaches/hard/digits-of-pi/DigitsOfPi.java
import java.util.Scanner;

public class DigitsOfPi {

    public static char getPiDecimalAt(int index) {
        String piDecimals = "14159265358979323846264338327950288419716939937510582097494459230781640628620899862803482534211706798214808651328230664709384460955058223172535940812848111745028410270938521105559644622948954930381964428";
        char decimal = piDecimals.charAt(index);
        return decimal;
    }

    public static void main(String[] args) {
        // - 1 to convert the index to 0-based index
        Scanner scanner = new Scanner(System.in);
        int index = scanner.nextInt() - 1;

        System.out.println(getPiDecimalAt(index));

        scanner.close();
    }

}
